### ISSUE TITLE
[Dataset requests] New datasets for Text Classification

### ISSUE THREAD

**Issue by thomwolf** [huggingface/nlp#353]
We are missing a few datasets for Text Classification which is an important field.

Namely, it would be really nice to add:
- Yelp-5
- Movie review (Movie Review (MR) dataset [156])
- SST (Stanford Sentiment Treebank)
- Multi-Perspective Question Answering (MPQA) dataset
- Amazon. This is a popular corpus of product reviews collected from the Amazon website [159]. It contains labels for both binary classification and multi-class (5-class) classification
- 20 Newsgroups. The 20 Newsgroups dataset
- Sogou News dataset
- Reuters news. The Reuters-21578 dataset [165]
- DBpedia. The DBpedia dataset [170]
- Ohsumed. The Ohsumed collection [171] is a subset of the MEDLINE database
- EUR-Lex. The EUR-Lex dataset
- WOS. The Web Of Science (WOS) dataset 
- PubMed. PubMed [173]
- TREC-QA. TREC-QA
- Quora. The Quora dataset [180]

All these datasets are cited in https://arxiv.org/abs/2004.03705

**Comment by thomwolf** [comment]
Pinging @mariamabarham as well

**Comment by jxmorris12** [comment]
- I added MR! It's called `rotten_tomatoes`
- SST is part of GLUE, no?
- I also added `ag_news`

**Comment by mariamabarham** [comment]
Thanks @jxmorris12 for pointing this out. 

In glue we only have SST-2 maybe we can add separately SST-1.

**Comment by jxmorris12** [comment]
This is the homepage for the Amazon dataset: https://www.kaggle.com/datafiniti/consumer-reviews-of-amazon-products

Is there an easy way to download kaggle datasets programmatically? If so, I can add this one!

**Comment by mariamabarham** [comment]
Hi @jxmorris12 for now I think our `dl_manager` does not download from Kaggle.
@thomwolf , @lhoestq

**Comment by ghomasHudson** [comment]
Pretty sure the quora dataset is the same one I implemented here: https://github.com/huggingface/nlp/pull/366

**Comment by moscow25** [comment]
Great list. Any idea if Amazon Reviews has been added?

- ~40 GB of text (sadly no emoji)
- popular MLM pre-training dataset before bigger datasets like WebText https://arxiv.org/abs/1808.01371
- turns out that binarizing the 1-5 star rating leads to great Pos/Neg/Neutral dataset, T5 paper claims to get very high accuracy (98%!) on this with small amount of finetuning https://arxiv.org/abs/2004.14546

Apologies if it's been included (great to see where) and if not, it's one of the better medium/large NLP dataset for semi-supervised learning, albeit a bit out of date. 

Thanks!! 

cc @sshleifer

**Comment by moscow25** [comment]
On the Amazon Reviews dataset, the original UCSD website has noted these are now updated to include product reviews through 2018 -- actually quite recent compared to many other datasets. Almost certainly the largest NLP dataset out there with labels!
https://jmcauley.ucsd.edu/data/amazon/ 

Any chance someone has time to onboard this dataset in a HF way?

cc @sshleifer

**Comment by johneckberg** [comment]
@albertvillanova How up to date is this issue? I see that some of these datasets are now on huggingface but have not been checked off the list

**Comment by YasserElHaddar16** [comment]
The OHSUMED dataset is already available as a community dataset at [community-datasets/ohsumed](https://huggingface.co/datasets/community-datasets/ohsumed).

The community version includes:
- Full dataset with 348,566 references from MEDLINE
- Train/test splits (54,709 / 293,855 examples)
- All original fields (title, abstract, MeSH terms, author, source, publication type)

Given this existing implementation, we might want to:
1. Mark OHSUMED as completed in the issue list
2. Focus on the remaining datasets (MPQA, EUR-Lex, PubMed)
3. Consider if any improvements to the community version are needed before potentially promoting it to the main library

**Comment by Yasserelhaddar** [comment]
The OHSUMED dataset is already available as a community dataset at [community-datasets/ohsumed](https://huggingface.co/datasets/community-datasets/ohsumed).

The community version includes:

- Full dataset with 348,566 references from MEDLINE
- Train/test splits (54,709 / 293,855 examples)
- All original fields (title, abstract, MeSH terms, author, source, publication type)

Given this existing implementation, we might want to Mark OHSUMED as completed in the issue list.

**Comment by Yasserelhaddar** [comment]
I notice that several datasets from this list are already available:

1. OHSUMED: Available as [community-datasets/ohsumed](https://huggingface.co/datasets/community-datasets/ohsumed)
2. EUR-Lex: Available in two versions:
   - [coastalcph/multi_eurlex](https://huggingface.co/datasets/coastalcph/multi_eurlex)
   - [NLP-AUEB/eurlex](https://huggingface.co/datasets/NLP-AUEB/eurlex)
3. PubMed: Available as [ncbi/pubmed](https://huggingface.co/datasets/ncbi/pubmed)

Is this issue not updated, or am I missing something? The only dataset from the remaining list that seems to truly need addition is the MPQA dataset.

**Comment by jxmorris12** [comment]
MPQA is definitely on the hub: https://huggingface.co/datasets/jxm/mpqa

**Comment by LeonxLJX** [comment]
I'd like to take this one (`[Dataset requests] New datasets for Text Classification`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)

**Comment by LeonxLJX** [comment]
I'd like to take this one (`[Dataset requests] New datasets for Text Classification`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)
